In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

plt.style.use('seaborn-v0_8')   #sets a clean visual style for all charts.
sns.set_palette("husl")

print("Libraries loaded successfully!")

Libraries loaded successfully!


In [3]:
df = pd.read_csv('data/DataCOSupplyChainDataset.csv',encoding='latin 1')

df.head()

,Type,Days for shipping (real),Days for shipment (scheduled),Benefit per order,Sales per customer,Delivery Status,Late_delivery_risk,Category Id,Category Name,Customer City,...,Order Zipcode,Product Card Id,Product Category Id,Product Description,Product Image,Product Name,Product Price,Product Status,shipping date (DateOrders),Shipping Mode
0,DEBIT,3,4,91.250000,314.640015,Advance shipping,0,73,Sporting Goods,Caguas,...,NaN,1360,73,NaN,http://images.acmesports.sports/Smart+watch,Smart watch,327.75,0,2/3/2018 22:56,Standard Class
1,TRANSFER,5,4,-249.089996,311.359985,Late delivery,1,73,Sporting Goods,Caguas,...,NaN,1360,73,NaN,http://images.acmesports.sports/Smart+watch,Smart watch,327.75,0,1/18/2018 12:27,Standard Class
2,CASH,4,4,-247.779999,309.720001,Shipping on time,0,73,Sporting Goods,San Jose,...,NaN,1360,73,NaN,http://images.acmesports.sports/Smart+watch,Smart watch,327.75,0,1/17/2018 12:06,Standard Class
3,DEBIT,3,4,22.860001,304.809998,Advance shipping,0,73,Sporting Goods,Los Angeles,...,NaN,1360,73,NaN,http://images.acmesports.sports/Smart+watch,Smart watch,327.75,0,1/16/2018 11:45,Standard Class
4,PAYMENT,2,4,134.210007,298.250000,Advance shipping,0,73,Sporting Goods,Caguas,...,NaN,1360,73,NaN,http://images.acmesports.sports/Smart+watch,Smart watch,327.75,0,1/15/2018 11:24,Standard Class


In [4]:
df.columns.tolist()


['Type',
 'Days for shipping (real)',
 'Days for shipment (scheduled)',
 'Benefit per order',
 'Sales per customer',
 'Delivery Status',
 'Late_delivery_risk',
 'Category Id',
 'Category Name',
 'Customer City',
 'Customer Country',
 'Customer Email',
 'Customer Fname',
 'Customer Id',
 'Customer Lname',
 'Customer Password',
 'Customer Segment',
 'Customer State',
 'Customer Street',
 'Customer Zipcode',
 'Department Id',
 'Department Name',
 'Latitude',
 'Longitude',
 'Market',
 'Order City',
 'Order Country',
 'Order Customer Id',
 'order date (DateOrders)',
 'Order Id',
 'Order Item Cardprod Id',
 'Order Item Discount',
 'Order Item Discount Rate',
 'Order Item Id',
 'Order Item Product Price',
 'Order Item Profit Ratio',
 'Order Item Quantity',
 'Sales',
 'Order Item Total',
 'Order Profit Per Order',
 'Order Region',
 'Order State',
 'Order Status',
 'Order Zipcode',
 'Product Card Id',
 'Product Category Id',
 'Product Description',
 'Product Image',
 'Product Name',
 'Product P

In [12]:
discounted = df[df['Order Item Discount Rate'] > 0]
non_discounted = df[df['Order Item Discount Rate'] == 0]

print(discounted['Benefit per order'].mean())
print(non_discounted['Benefit per order'].mean())

21.699037319930934
26.66657382708536


Hypothesis testing: 

H₀ - Null hypothesis - There is no difference in profit between orders with discount and without discount - True untill proven wrong by stats and data

H₁ - Aternative Hypothesis - there might be something, there might be a difference - My accusation

α  - threshold value, if p-value is less than this, then null hypothesis Ho is proven false = 0.05 = probablitiy lower than 5% that difference is random noise 


In [17]:
from scipy.stats import ttest_ind 

t_val,p_val = ttest_ind(discounted['Benefit per order'], non_discounted['Benefit per order'], equal_var = False)

print(f'T value : {t_val}')
print(f'P Value : {p_val}')

T value : -4.380098734129683
P Value : 1.1971395363095526e-05


In [22]:
from math import sqrt

mean1 = discounted['Benefit per order'].mean()
mean2 = non_discounted['Benefit per order'].mean()
n1 = len(discounted)
n2 = len(non_discounted)
s1 = discounted['Benefit per order'].std()
s2 = non_discounted['Benefit per order'].std()

pooled_std = sqrt( ((n1-1)*s1**2 + (n2-1)*s2**2) / (n1 + n2 - 2) )

d = (mean1 - mean2) / pooled_std

print(f'Cohen d = {d.round(3)}')

Cohen d = -0.048


We reject/accept H₀ because p = 0.0001 which is below α = 0.05.
Discounted orders averaged $21.6 profit vs $26.6 for non-discounted.
However, Cohen's d = -0.048 which indicates a insignificant effect size.
This means we can conclude that there is difference from null hypothesis, discount negatively impacts the benefit per order, but even though the p value is less, it just gaurentees that there is impact from discounts nut our d-value which is way lesser than 0.2 hence the impact is really low on benefit.